Bulldozer.CSV/CSVComponent.cs

```diff
@@ -488,804 +820,2218 @@ public static int GetAttributeFieldType( string attributeTypeString )
 
-        private const int GroupTypeId = 0;                  /* String | Int [Requred] */
-        private const int GroupTypeName = 1;                /* String [Requred] */
-        private const int GroupTypeCreatedDate = 2;         /* Date (mm/dd/yyyy) [Optional] */
-        private const int GroupTypePurpose = 3;             /* Guid | Int | String [Optional] */
-        private const int GroupTypeInheritedGroupType = 4;  /* Guid | Int | String [Optional] */
-        private const int GroupTypeTakesAttendance = 5;     /* "YES" | "NO" [Optional] */
-        private const int GroupTypeWeekendService = 6;      /* "YES" | "NO" [Optional] */
-        private const int GroupTypeShowInGroupList = 7;     /* "YES" | "NO" [Optional] */
-        private const int GroupTypeShowInNav = 8;           /* "YES" | "NO" [Optional] */
-        private const int GroupTypeParentId = 9;            /* String | Int [Optional] */
-        private const int GroupTypeSelfReference = 10;      /* "YES" | "NO" [Optional] */
-        private const int GroupTypeWeeklySchedule = 11;     /* "YES" | "NO" [Optional] */
-        private const int GroupTypeDescription = 12;        /* String [Optional] */
+            CampusCache.Clear();
+
+            var campusesToCreate = importCampuses.Where( a => !CampusCache.All().Any( c => c.ForeignKey != null && c.ForeignKey == string.Format( "{0}^{1}", this.ImportInstanceFKPrefix, a.CampusId ) ) );
+            if ( campusesToCreate.Count() > 0 )
+            {
+                ReportProgress( 0, string.Format( "Creating {0} new Campuses...", campusesToCreate.Count() ) );
+            }
+
+            // Rock has a Unique Constraint on Campus.Name so, make sure campus name is unique and rename it if a new campus happens to have the same name as an existing campus
+            var usedCampusNames = CampusCache.All().Select( a => a.Name ).ToList();
+
+            foreach ( var importCampus in campusesToCreate )
+            {
+                var newCampus = new Campus()
+                {
+                    ForeignId = importCampus.CampusId.AsIntegerOrNull(),
+                    ForeignKey = string.Format( "{0}^{1}", ImportInstanceFKPrefix, importCampus.CampusId ),
+                    IsActive = true,
+                    Name = importCampus.CampusName,
+                    Guid = Guid.NewGuid()
+                };
+
+                if ( usedCampusNames.Any( a => a.Equals( importCampus.CampusName ) ) )
+                {
+                    newCampus.Name = importCampus.CampusName + $" ({ImportInstanceFKPrefix})";
+                }
+
+                usedCampusNames.Add( newCampus.Name );
+                campusService.Add( newCampus );
+                rockContext.SaveChanges();
+            }
+
+            CampusCache.Clear();
+            LoadCampusDict( rockContext );
+        }
+
+        /// <summary>
+        /// Add any GroupTypes that aren't in Rock yet
+        /// </summary>
+        private void AddGroupTypes()
```

Comment:
Bump against old "LoadGroupType" method to make sure all features are accounted for. Namely default role